Paygroups and Roles Page

alainalouisepayawal opened the pull request:
This PR is about the CRUD operations in the Paygroups and Roles Page.

Paygroups Page
![paygroups-page](https://github.com/torodeveloper/payroll-system-bellini-app/assets/158266869/7dfd44ce-2e04-4d4e-88cb-b9cb6ad5a017)

Paygroups Page - Create Modal
![paygroups-page-create](https://github.com/torodeveloper/payroll-system-bellini-app/assets/158266869/0b07ed7a-09d2-409f-9643-b05df6b23956)

Paygroups Page - Update Modal
![paygroups-page-update](https://github.com/torodeveloper/payroll-system-bellini-app/assets/158266869/baf4e0e6-03e6-481a-88bf-dc93c70f70ef)

Paygroups Page - Delete Modal
![paygroups-page-delete](https://github.com/torodeveloper/payroll-system-bellini-app/assets/158266869/6bc0e267-0a29-4110-88fa-5a1ed9923bf7)

alainalouisepayawal committed:
add pay groups and roles page

alainalouisepayawal committed:
CRUD operations for Roles Page

torodeveloper reviewed (changes requested):
<img width="1154" alt="image" src="https://github.com/torodeveloper/payroll-system-bellini-app/assets/121865290/ca68040f-2767-4ef6-b8b9-3f8b55b9b22b">

1. For the Shift In and Out you can use the datetime picker component
2. Disable the OK button if the form data is not yet completed.
3. Change the OK button label to "Create" and "Update" when updating.

torodeveloper reviewed (changes requested):
<img width="421" alt="image" src="https://github.com/torodeveloper/payroll-system-bellini-app/assets/121865290/dd15ae99-a2ed-45f7-a7ea-db1984b1c823">

Here, instead of changing the color of buttons one by one, let's add a theme to the app config and change the buttons theme there.

torodeveloper reviewed (changes requested):
![2024-04-17 14-10-59 2024-04-17 14_13_44](https://github.com/torodeveloper/payroll-system-bellini-app/assets/121865290/91b251d4-68f5-443d-ab29-6bb8c88f2cb1)

The Roles pagination started to display the incorrect items after navigating through pages.

torodeveloper reviewed (changes requested):
<img width="756" alt="image" src="https://github.com/torodeveloper/payroll-system-bellini-app/assets/121865290/a6daafb9-3596-43b2-99f1-b78a593c8750">

Shouldn't this be a number input?

torodeveloper reviewed (changes requested):
<img width="1045" alt="image" src="https://github.com/torodeveloper/payroll-system-bellini-app/assets/121865290/a7dede40-b647-47c8-906b-77f705bc467b">

The Update button gets stuck on loading when I click on it while there are no selections in the table. It should be disabled in that case. Same with the Delete button.

torodeveloper commented:
<img width="1243" alt="image" src="https://github.com/torodeveloper/payroll-system-bellini-app/assets/121865290/e52f92ee-b9a9-4e0b-8f5a-a786e9a1e252">

What is the purpose of wrapping the core Navigation component with a custom component? Is it for the padding only?

Commit:
latest changes in paygroup, roles and benefits page

alainalouisepayawal committed:
update on navbar and benefits page

alainalouisepayawal commented:
> <img alt="image" width="1154" src="https://private-user-images.githubusercontent.com/121865290/323096781-ca68040f-2767-4ef6-b8b9-3f8b55b9b22b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NTIwODUsIm5iZiI6MTcxMzc1MTc4NSwicGF0aCI6Ii8xMjE4NjUyOTAvMzIzMDk2NzgxLWNhNjgwNDBmLTI3NjctNGVmNi1iOGI5LTNmOGI1NWI5YjIyYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyMlQwMjA5NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jYTQzODVmZjg2ZDFmZWIyYmI1ZjAxNGFhNzhiMjJkZGFkOGUyYjViYzY3OTFkNTM0MDE4NWI4MWViYjYyY2Q1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.tywkJ-jt06lmPaKdLen10VNhD1rIzzBnrPNDqEHALyA">
> 1. For the Shift In and Out you can use the datetime picker component
> 2. Disable the OK button if the form data is not yet completed.
> 3. Change the OK button label to "Create" and "Update" when updating.

----------------------------
1. Changed the Shift In and Shift Out input into Datetime Picker component with 12h format.
![image](https://github.com/torodeveloper/payroll-system-bellini-app/assets/158266869/ba38af58-c7f9-492a-af0d-8603e0440000)

2. Created a function to validate data on change and will only enable the button if the data needed for the payload is complete, hence it will stay disabled.
![image](https://github.com/torodeveloper/payroll-system-bellini-app/assets/158266869/5d33968c-b0da-4040-a600-d4f0164e4e69)

![image](https://github.com/torodeveloper/payroll-system-bellini-app/assets/158266869/03550241-7c5f-4a22-8c22-fe9370bf8d3d)

3. Changed the button labels from “OK” to “Create” and “Update” accordingly.
![image](https://github.com/torodeveloper/payroll-system-bellini-app/assets/158266869/5a8d5f74-0276-4905-89e4-cede18d58e9a)

![image](https://github.com/torodeveloper/payroll-system-bellini-app/assets/158266869/9f9f053a-cea7-41d1-a74d-c1cdd3bd8655)

alainalouisepayawal commented:
> <img alt="image" width="421" src="https://private-user-images.githubusercontent.com/121865290/323099347-dd15ae99-a2ed-45f7-a7ea-db1984b1c823.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NTIwODUsIm5iZiI6MTcxMzc1MTc4NSwicGF0aCI6Ii8xMjE4NjUyOTAvMzIzMDk5MzQ3LWRkMTVhZTk5LWEyZWQtNDVmNy1hN2VhLWRiMTk4NGIxYzgyMy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyMlQwMjA5NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hYmIwZDY5NWI1YjdhY2IwZmEyMjgyYjNmOTJiOWU2MmQzNmZiODc2OTlhN2QyZjU2OTRhNzI5YWZhNGE5NmUzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.3ZxZAo-jKRrjzDQCykXn4CZXFT2ZKhFBlMvqXVX1-Bo">
> Here, instead of changing the color of buttons one by one, let's add a theme to the app config and change the buttons theme there.

-------------------
1. Utilized the Theme feature of Bellini app config.
![image](https://github.com/torodeveloper/payroll-system-bellini-app/assets/158266869/bcb23a4f-f072-4016-8d70-8db0d97f18f1)

alainalouisepayawal commented:
> ![2024-04-17 14-10-59 2024-04-17 14_13_44](https://private-user-images.githubusercontent.com/121865290/323100882-91b251d4-68f5-443d-ab29-6bb8c88f2cb1.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NTIwODUsIm5iZiI6MTcxMzc1MTc4NSwicGF0aCI6Ii8xMjE4NjUyOTAvMzIzMTAwODgyLTkxYjI1MWQ0LTY4ZjUtNDQzZC1hYjI5LTZiYjhjODhmMmNiMS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyMlQwMjA5NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wMWFhZTRmMGNjZDNiYThkMjE1YjZjODYxNjk2YjI0MzMzOWNiMzA3NjhiOTk2YmFkMmFmZWMzNTg0NDdiOWI0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.RTmPO5C2UthUu3Zz1_gH33XoodMOgpkT4KQ4BoWxgJk) [ ![2024-04-17 14-10-59 2024-04-17 14_13_44](https://private-user-images.githubusercontent.com/121865290/323100882-91b251d4-68f5-443d-ab29-6bb8c88f2cb1.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NTIwODUsIm5iZiI6MTcxMzc1MTc4NSwicGF0aCI6Ii8xMjE4NjUyOTAvMzIzMTAwODgyLTkxYjI1MWQ0LTY4ZjUtNDQzZC1hYjI5LTZiYjhjODhmMmNiMS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyMlQwMjA5NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wMWFhZTRmMGNjZDNiYThkMjE1YjZjODYxNjk2YjI0MzMzOWNiMzA3NjhiOTk2YmFkMmFmZWMzNTg0NDdiOWI0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.RTmPO5C2UthUu3Zz1_gH33XoodMOgpkT4KQ4BoWxgJk) ](https://private-user-images.githubusercontent.com/121865290/323100882-91b251d4-68f5-443d-ab29-6bb8c88f2cb1.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NTIwODUsIm5iZiI6MTcxMzc1MTc4NSwicGF0aCI6Ii8xMjE4NjUyOTAvMzIzMTAwODgyLTkxYjI1MWQ0LTY4ZjUtNDQzZC1hYjI5LTZiYjhjODhmMmNiMS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyMlQwMjA5NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wMWFhZTRmMGNjZDNiYThkMjE1YjZjODYxNjk2YjI0MzMzOWNiMzA3NjhiOTk2YmFkMmFmZWMzNTg0NDdiOWI0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.RTmPO5C2UthUu3Zz1_gH33XoodMOgpkT4KQ4BoWxgJk) [ ](https://private-user-images.githubusercontent.com/121865290/323100882-91b251d4-68f5-443d-ab29-6bb8c88f2cb1.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NTIwODUsIm5iZiI6MTcxMzc1MTc4NSwicGF0aCI6Ii8xMjE4NjUyOTAvMzIzMTAwODgyLTkxYjI1MWQ0LTY4ZjUtNDQzZC1hYjI5LTZiYjhjODhmMmNiMS5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyMlQwMjA5NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wMWFhZTRmMGNjZDNiYThkMjE1YjZjODYxNjk2YjI0MzMzOWNiMzA3NjhiOTk2YmFkMmFmZWMzNTg0NDdiOWI0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.RTmPO5C2UthUu3Zz1_gH33XoodMOgpkT4KQ4BoWxgJk)
> 
> The Roles pagination started to display the incorrect items after navigating through pages.

--------------------
1. Fixed the pagination function, specifically the “On Page Change.”
![pagination](https://github.com/torodeveloper/payroll-system-bellini-app/assets/158266869/6ecb63c0-da99-4f07-b720-8abc7d9bb907)

alainalouisepayawal commented:
> <img alt="image" width="756" src="https://private-user-images.githubusercontent.com/121865290/323104157-a6daafb9-3596-43b2-99f1-b78a593c8750.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NTIwODUsIm5iZiI6MTcxMzc1MTc4NSwicGF0aCI6Ii8xMjE4NjUyOTAvMzIzMTA0MTU3LWE2ZGFhZmI5LTM1OTYtNDNiMi05OWYxLWI3OGE1OTNjODc1MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyMlQwMjA5NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mYzE0NmRhY2E0MTNlNzVmMmNhOTM5NTZjMTY1MTdkZjFlZDlhODNkMGRmMjljMTMxZGZhYTYxNmI2NzMxZDc3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.CbB8zxgj4T-w-OLyF_x6Xv1Ct8qQXJ5awvycf5Rq40Y">
> Shouldn't this be a number input?

-----------------
1. Changed the text input into number input for data that accepts number instead of string.
overtimeMultiplier: number             requiredHours: number
payCurrency: string                            roleTitle: string
payGroupName: string                      shiftIn: time
payRate: number                                shiftOut: time

![image](https://github.com/torodeveloper/payroll-system-bellini-app/assets/158266869/3735c96c-42a9-4aff-8e77-7927075a7efd)

alainalouisepayawal commented:
> <img alt="image" width="1045" src="https://private-user-images.githubusercontent.com/121865290/323105142-a7dede40-b647-47c8-906b-77f705bc467b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NTIwODUsIm5iZiI6MTcxMzc1MTc4NSwicGF0aCI6Ii8xMjE4NjUyOTAvMzIzMTA1MTQyLWE3ZGVkZTQwLWI2NDctNDdjOC05MDZiLTc3ZjcwNWJjNDY3Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyMlQwMjA5NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MzQxN2M2MDVlZjVhNmI1YWQ3NDdlMjY3Njg1OWMzMmVjZWUzODY5NDJhOGVlYjEwYjM0ZjgyZWEyZWZhZGFlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.J9OCxYi_ffJm3u3vSc9dVmnVGu-ef0hyAWLB5h3huuA">
> The Update button gets stuck on loading when I click on it while there are no selections in the table. It should be disabled in that case. Same with the Delete button.

------------------
1. Disabled the Update and Delete button if a row/item is not selected.
![image](https://github.com/torodeveloper/payroll-system-bellini-app/assets/158266869/35f479c3-37cd-4943-9491-d2cf53384d4f)

alainalouisepayawal commented:
> <img alt="image" width="1243" src="https://private-user-images.githubusercontent.com/121865290/323109050-e52f92ee-b9a9-4e0b-8f5a-a786e9a1e252.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NTIwODUsIm5iZiI6MTcxMzc1MTc4NSwicGF0aCI6Ii8xMjE4NjUyOTAvMzIzMTA5MDUwLWU1MmY5MmVlLWI5YTktNGUwYi04ZjVhLWE3ODZlOWExZTI1Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyMlQwMjA5NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05MDIzZmE1MDc4YzgxMjRkN2I0YWE4OGZhZTdkMDZhYjZhNzJhNjBkZjIyMjY3YWU2OGM1MTc5Nzk0OTI4ODkwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.XRP0GXOel14m77DC_rSoaGOVwjm7s9Tbyp4iPLGbvJM">
> What is the purpose of wrapping the core Navigation component with a custom component? Is it for the padding only?

-----------------
1. Unwrapped the core navigation component from the div and added the padding using CSS.
![image](https://github.com/torodeveloper/payroll-system-bellini-app/assets/158266869/43bb4ea1-c6b2-46e1-94cc-1f55ca40f05e)

alainalouisepayawal committed:
fixed validateData

alainalouisepayawal committed:
fixed bugs on handleUpdate function

alainalouisepayawal committed:
Update README.md

torodeveloper commented:
> > <img alt="image" width="1045" src="https://private-user-images.githubusercontent.com/121865290/323105142-a7dede40-b647-47c8-906b-77f705bc467b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NTIwODUsIm5iZiI6MTcxMzc1MTc4NSwicGF0aCI6Ii8xMjE4NjUyOTAvMzIzMTA1MTQyLWE3ZGVkZTQwLWI2NDctNDdjOC05MDZiLTc3ZjcwNWJjNDY3Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyMlQwMjA5NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MzQxN2M2MDVlZjVhNmI1YWQ3NDdlMjY3Njg1OWMzMmVjZWUzODY5NDJhOGVlYjEwYjM0ZjgyZWEyZWZhZGFlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.J9OCxYi_ffJm3u3vSc9dVmnVGu-ef0hyAWLB5h3huuA">
> > The Update button gets stuck on loading when I click on it while there are no selections in the table. It should be disabled in that case. Same with the Delete button.
> 
> 1. Disabled the Update and Delete button if a row/item is not selected.
>    ![image](https://private-user-images.githubusercontent.com/158266869/324308447-35f479c3-37cd-4943-9491-d2cf53384d4f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NjMzODcsIm5iZiI6MTcxMzc2MzA4NywicGF0aCI6Ii8xNTgyNjY4NjkvMzI0MzA4NDQ3LTM1ZjQ3OWMzLTM3Y2QtNDk0My05NDkxLWQyY2Y1MzM4NGQ0Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyMlQwNTE4MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NWEyMjQxYmJmMDgzMGFkY2E4ZGVmYjkzODEwNjZkNmVlMWM2ZjMxODcxMDI4YzkzMzEwMDhhNzQxZTM5YjQ5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.OlvP2LuSeg3FI0jiV2kPSGG5C4Syt8sIv42DX04X6Jo)

![2024-04-22 13-20-40 2024-04-22 13_22_41](https://github.com/torodeveloper/payroll-system-bellini-app/assets/121865290/30ffe1c5-74bf-4e53-b007-706065bcfe3f)
Removing the selection doesn't disable the buttons again.

torodeveloper commented on components/new-role/stylesheets/new-role.css:
Is this CSS file unused?

alainalouisepayawal replied:
Yes Sir, I forgot to delete it. I apologize for the oversight.

torodeveloper commented on stylesheets/paygroupsPage.css:
Let's stick to kebab case file naming.

torodeveloper replied:
We don't have to redo it on every page. Rename this to `table-reset.css` then include all the pages to it:

```css
.paygroups-table,
.roles-table {
    text-align: center;
    overflow-y: auto;
}

etc...
```

alainalouisepayawal replied:
> We don't have to redo it on every page. Rename this to `table-reset.css` then include all the pages to it:
> 
> ```css
> .paygroups-table,
> .roles-table {
>     text-align: center;
>     overflow-y: auto;
> }
> 
> etc...
> ```

----------------
1. Renamed the css to 'table-reset.css'
![image](https://github.com/torodeveloper/payroll-system-bellini-app/assets/158266869/1072d46d-4bac-44ad-adb6-639a1e2c9f80)

torodeveloper replied:
That is not the right selector. It should be separated by a comma per page:

```css
.paygroups-table,
.roles-table,
.benifits-table { ... }

.pay-groups-table::webkit-scrollbar,
.roles-table::webkit-scrollbar,
.benifits-table::webkit-scrollbar { ... }

...
```

alainalouisepayawal replied:
1. Changed the declaration of classes in table-reset.css in order to apply the styles.
![image](https://github.com/torodeveloper/payroll-system-bellini-app/assets/158266869/d0883bd5-96ea-491c-8b87-12ca95ac7345)

alainalouisepayawal commented:
> > > <img alt="image" width="1045" src="https://private-user-images.githubusercontent.com/121865290/323105142-a7dede40-b647-47c8-906b-77f705bc467b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NTIwODUsIm5iZiI6MTcxMzc1MTc4NSwicGF0aCI6Ii8xMjE4NjUyOTAvMzIzMTA1MTQyLWE3ZGVkZTQwLWI2NDctNDdjOC05MDZiLTc3ZjcwNWJjNDY3Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyMlQwMjA5NDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MzQxN2M2MDVlZjVhNmI1YWQ3NDdlMjY3Njg1OWMzMmVjZWUzODY5NDJhOGVlYjEwYjM0ZjgyZWEyZWZhZGFlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.J9OCxYi_ffJm3u3vSc9dVmnVGu-ef0hyAWLB5h3huuA">
> > > The Update button gets stuck on loading when I click on it while there are no selections in the table. It should be disabled in that case. Same with the Delete button.
> > 
> > 
> > 
> > 1. Disabled the Update and Delete button if a row/item is not selected.
> >    ![image](https://private-user-images.githubusercontent.com/158266869/324308447-35f479c3-37cd-4943-9491-d2cf53384d4f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NjMzODcsIm5iZiI6MTcxMzc2MzA4NywicGF0aCI6Ii8xNTgyNjY4NjkvMzI0MzA4NDQ3LTM1ZjQ3OWMzLTM3Y2QtNDk0My05NDkxLWQyY2Y1MzM4NGQ0Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyMlQwNTE4MDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NWEyMjQxYmJmMDgzMGFkY2E4ZGVmYjkzODEwNjZkNmVlMWM2ZjMxODcxMDI4YzkzMzEwMDhhNzQxZTM5YjQ5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.OlvP2LuSeg3FI0jiV2kPSGG5C4Syt8sIv42DX04X6Jo)
> 
> ![2024-04-22 13-20-40 2024-04-22 13_22_41](https://private-user-images.githubusercontent.com/121865290/324337301-30ffe1c5-74bf-4e53-b007-706065bcfe3f.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NzAwOTcsIm5iZiI6MTcxMzc2OTc5NywicGF0aCI6Ii8xMjE4NjUyOTAvMzI0MzM3MzAxLTMwZmZlMWM1LTc0YmYtNGU1My1iMDA3LTcwNjA2NWJjZmUzZi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyMlQwNzA5NTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zM2Y0NGY1NTJmMWViNDgwNzY0MjA0M2IzMTJlNWNiMzNmYjRlZTY4Y2EwOTA3NTQ0YTU3YmZiYzFjYzFmYzEzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.UFDk3vMtrCHR7XlNA7ZMCcpjfc3a_cA7ZF2HU5oD5TA) [ ![2024-04-22 13-20-40 2024-04-22 13_22_41](https://private-user-images.githubusercontent.com/121865290/324337301-30ffe1c5-74bf-4e53-b007-706065bcfe3f.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NzAwOTcsIm5iZiI6MTcxMzc2OTc5NywicGF0aCI6Ii8xMjE4NjUyOTAvMzI0MzM3MzAxLTMwZmZlMWM1LTc0YmYtNGU1My1iMDA3LTcwNjA2NWJjZmUzZi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyMlQwNzA5NTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zM2Y0NGY1NTJmMWViNDgwNzY0MjA0M2IzMTJlNWNiMzNmYjRlZTY4Y2EwOTA3NTQ0YTU3YmZiYzFjYzFmYzEzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.UFDk3vMtrCHR7XlNA7ZMCcpjfc3a_cA7ZF2HU5oD5TA) ](https://private-user-images.githubusercontent.com/121865290/324337301-30ffe1c5-74bf-4e53-b007-706065bcfe3f.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NzAwOTcsIm5iZiI6MTcxMzc2OTc5NywicGF0aCI6Ii8xMjE4NjUyOTAvMzI0MzM3MzAxLTMwZmZlMWM1LTc0YmYtNGU1My1iMDA3LTcwNjA2NWJjZmUzZi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyMlQwNzA5NTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zM2Y0NGY1NTJmMWViNDgwNzY0MjA0M2IzMTJlNWNiMzNmYjRlZTY4Y2EwOTA3NTQ0YTU3YmZiYzFjYzFmYzEzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.UFDk3vMtrCHR7XlNA7ZMCcpjfc3a_cA7ZF2HU5oD5TA) [ ](https://private-user-images.githubusercontent.com/121865290/324337301-30ffe1c5-74bf-4e53-b007-706065bcfe3f.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NzAwOTcsIm5iZiI6MTcxMzc2OTc5NywicGF0aCI6Ii8xMjE4NjUyOTAvMzI0MzM3MzAxLTMwZmZlMWM1LTc0YmYtNGU1My1iMDA3LTcwNjA2NWJjZmUzZi5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyMlQwNzA5NTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zM2Y0NGY1NTJmMWViNDgwNzY0MjA0M2IzMTJlNWNiMzNmYjRlZTY4Y2EwOTA3NTQ0YTU3YmZiYzFjYzFmYzEzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.UFDk3vMtrCHR7XlNA7ZMCcpjfc3a_cA7ZF2HU5oD5TA) Removing the selection doesn't disable the buttons again.

----------------
1. Modified the selectPaygroup, selectRole and selectBenefit in order to fix the issue.

**Paygroup Page**
![selectPaygroup](https://github.com/torodeveloper/payroll-system-bellini-app/assets/158266869/21a96c91-d65d-4de0-8bbb-1b2ac168dda3)

**Roles Page**
![selectRole](https://github.com/torodeveloper/payroll-system-bellini-app/assets/158266869/d9895cc1-54bd-48f9-9958-e543d35e0875)

**Benefits Page**
![selectBenefit](https://github.com/torodeveloper/payroll-system-bellini-app/assets/158266869/98ffbada-4f79-4116-87ef-117bb5b3b321)

alainalouisepayawal committed:
latest bug fix

alainalouisepayawal committed:
delete benefits page and benefits modals
